consensus/src/creation/mod.rs

```diff
@@ -55,16 +55,23 @@ async fn create_unit<H: Hasher>(
     round: Round,
     creator: &mut Creator<H>,
     create_lag: &DelaySchedule,
-    mut can_create: bool,
+    mut try_create: bool,
     incoming_parents: &mut Receiver<Unit<H>>,
     mut exit: &mut oneshot::Receiver<()>,
 ) -> Result<(PreUnit<H>, Vec<H::Hash>), ()> {
-    let mut delay = Delay::new(create_lag(round.into())).fuse();
+    const THIRTY_MINUTES: Duration = Duration::from_secs(30 * 60);
+
+    let initial_delay = match try_create {
```

Comment:
I would rather follow @DamianStraszak 's solution and then it's somewhat more natural to set the initial delay to zero if `skip_delay` is true.

This would also remove the new named constant, although maybe it should stay... anyway, if it stays I would suggest a different name for it, maybe something like `VERY_LONG_TIME`, 'cause the whole idea is that this should never pass after the first time, and the fact that it's set to 30mins is only a bit of a coincidence.

Reply:
I strongly prefer a zero delay over an `Option`, especially because of the match at the beginning of the method becoming completely unnecessary.